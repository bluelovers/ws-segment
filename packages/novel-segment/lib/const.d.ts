export declare const enum EnumDictDatabase {
    SYNONYM = "SYNONYM",
    TABLE = "TABLE",
    STOPWORD = "STOPWORD",
    /**
     * 字典黑名單 在主字典內刪除此字典內有的條目
     */
    BLACKLIST = "BLACKLIST",
    /**
     * 優化器黑名單 會防止部分優化器去組合此字典內的詞
     * 例如 人名 自動組合之類
     */
    BLACKLIST_FOR_OPTIMIZER = "BLACKLIST_FOR_OPTIMIZER",
    /**
     * 轉換黑名單 動態轉換字詞時會忽略此字典內的詞
     */
    BLACKLIST_FOR_SYNONYM = "BLACKLIST_FOR_SYNONYM"
}
