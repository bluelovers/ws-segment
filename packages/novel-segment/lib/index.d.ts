/**
 * Created by user on 2018/4/16/016.
 */
import getDefaultModList from './mod';
import { Segment } from './Segment';
import { useDefault } from './defaults';
export { getDefaultModList };
export { Segment };
export { useDefault };
export default Segment;
