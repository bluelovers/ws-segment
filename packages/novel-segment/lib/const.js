"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumDictDatabase = void 0;
var EnumDictDatabase;
(function (EnumDictDatabase) {
    EnumDictDatabase["SYNONYM"] = "SYNONYM";
    EnumDictDatabase["TABLE"] = "TABLE";
    EnumDictDatabase["STOPWORD"] = "STOPWORD";
    /**
     * 字典黑名單 在主字典內刪除此字典內有的條目
     */
    EnumDictDatabase["BLACKLIST"] = "BLACKLIST";
    /**
     * 優化器黑名單 會防止部分優化器去組合此字典內的詞
     * 例如 人名 自動組合之類
     */
    EnumDictDatabase["BLACKLIST_FOR_OPTIMIZER"] = "BLACKLIST_FOR_OPTIMIZER";
    /**
     * 轉換黑名單 動態轉換字詞時會忽略此字典內的詞
     */
    EnumDictDatabase["BLACKLIST_FOR_SYNONYM"] = "BLACKLIST_FOR_SYNONYM";
})(EnumDictDatabase = exports.EnumDictDatabase || (exports.EnumDictDatabase = {}));
//# sourceMappingURL=const.js.map