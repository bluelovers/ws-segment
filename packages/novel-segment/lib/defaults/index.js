"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useDefault = void 0;
const mods_1 = require("./mods");
const dict_1 = require("./dict");
function useDefault(segment, options = {}) {
    // 识别模块
    !options.nomod && mods_1.useDefaultMods(segment, options);
    // 字典文件
    !options.nodict && dict_1.useDefaultDicts(segment, options);
    return segment;
}
exports.useDefault = useDefault;
//# sourceMappingURL=index.js.map