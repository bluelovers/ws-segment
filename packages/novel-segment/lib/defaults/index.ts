/**
 * Created by user on 2019/6/26.
 */
import { Segment } from '../Segment';
import { useDefaultMods } from './mods';
import { useDefaultDicts } from './dict';

/**
 * @private
 */
export interface IUseDefaultOptionsDicts
{
	/**
	 * 不載入 字典
	 */
	nodict?: boolean,
	/**
	 * 載入 node-novel 相關字典
	 */
	nodeNovelMode?: boolean,
}

/**
 * @private
 */
export interface IUseDefaultOptionsMods
{
	all_mod?: boolean,
	nomod?: boolean,
}

export interface IUseDefaultOptions extends IUseDefaultOptionsDicts, IUseDefaultOptionsMods
{

}

export function useDefault(segment: Segment, options: IUseDefaultOptions = {})
{

	// 识别模块
	!options.nomod && useDefaultMods(segment, options);

	// 字典文件
	!options.nodict && useDefaultDicts(segment, options);

	return segment;
}
