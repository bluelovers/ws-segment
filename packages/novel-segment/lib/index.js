"use strict";
/**
 * Created by user on 2018/4/16/016.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useDefault = exports.Segment = exports.getDefaultModList = void 0;
const tslib_1 = require("tslib");
const mod_1 = tslib_1.__importDefault(require("./mod"));
exports.getDefaultModList = mod_1.default;
const Segment_1 = require("./Segment");
Object.defineProperty(exports, "Segment", { enumerable: true, get: function () { return Segment_1.Segment; } });
const defaults_1 = require("./defaults");
Object.defineProperty(exports, "useDefault", { enumerable: true, get: function () { return defaults_1.useDefault; } });
exports.default = Segment_1.Segment;
//# sourceMappingURL=index.js.map