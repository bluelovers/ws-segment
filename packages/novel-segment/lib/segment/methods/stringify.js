"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringify = void 0;
/**
 * 将单词数组连接成字符串
 *
 * @param {Array} words 单词数组
 * @return {String}
 */
function stringify(words, ...argv) {
    return words.map(function (item) {
        if (typeof item === 'string') {
            return item;
        }
        else if ('w' in item) {
            return item.w;
        }
        else {
            throw new TypeError(`not a valid segment result list`);
        }
    }).join('');
}
exports.stringify = stringify;
//# sourceMappingURL=stringify.js.map