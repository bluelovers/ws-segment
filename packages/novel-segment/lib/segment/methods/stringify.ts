import { IWord } from '../types';

/**
 * 将单词数组连接成字符串
 *
 * @param {Array} words 单词数组
 * @return {String}
 */
export function stringify(words: Array<IWord | string>, ...argv): string
{
	return words.map(function (item)
	{
		if (typeof item === 'string')
		{
			return item;
		}
		else if ('w' in item)
		{
			return item.w;
		}
		else
		{
			throw new TypeError(`not a valid segment result list`)
		}
	}).join('');
}
