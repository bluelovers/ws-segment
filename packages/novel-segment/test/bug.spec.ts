//@noUnusedParameters:false

import { basename, extname } from 'path';
import { createSegment } from './lib/index';
import { stringifyList, stringify } from '@novel-segment/stringify';

describe(`bug`, () =>
{
	const segment = createSegment(true, {
		nodeNovelMode: true,
	});

	test(`check word is constructor`, () =>
	{
		let words = segment.doSegment(`inspection.dead.code.problem.synopsis28.constructor=构造函数有一个用法,但它是不可到达的从入口点.`);

		let actual = stringify(words);

		expect(actual).toContain(`inspection.dead.code.problem.synopsis28.constructor`);
		expect(actual).not.toContain(`[native code]`);

		expect(actual).toMatchSnapshot();

	});

})
