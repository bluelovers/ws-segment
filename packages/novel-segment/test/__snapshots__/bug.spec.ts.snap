// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bug check word is constructor 1`] = `"inspection.dead.code.problem.synopsis28.constructor=构造函数有一个用法,但它是不可到达的从入口点."`;
